File: pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;

@Controller
@ResponseBody
@RequestMapping("/brand")
public class BrandController {

	@Reference
	private BrandService brandService;
	
	
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand tbBrand,@RequestParam(name="currentPage")int currentPage,@RequestParam(name="pageSize")int pageSize) {
		PageResult findByPage = brandService.findByPage(tbBrand, currentPage, pageSize);
		
		return findByPage;
	}
	
	@RequestMapping("/delete")
	public Result delete(long[] ids) {
		try {
			brandService.delete(ids);
			return new Result(true,"删除成功！");
		} catch (Exception e) {
			// TODO: handle exception
			return new Result(false,"删除失败");
		}
	}
	
	@RequestMapping("/add")
	public Result add(@RequestBody TbBrand brand) {
		try {
			brandService.add(brand);
			return new Result(true,"新增成功");
		} catch (Exception e) {
			// TODO: handle exception
			e.printStackTrace();
			return new Result(false,"新增失败");
		}
	}
	
	@RequestMapping("/findAll")
	public List<TbBrand> findAll() {
		return brandService.findAll();
	}
	
	@RequestMapping("/findByPage")
	public PageResult findByPage(int currentPage,int pageSize) {
		
		PageResult pageResult = brandService.findByPage(currentPage, pageSize);
		return pageResult;
	}
	
	@RequestMapping("/findById")
	public TbBrand findById(long id) {
		TbBrand findById = brandService.findById(id);
		return findById;
	}
	
	@RequestMapping("/update")
	public Result update(@RequestBody TbBrand brand) {
		try {
			brandService.update(brand);
			return new Result(true,"修改成功");
		} catch (Exception e) {
			// TODO: handle exception
			return new Result(false,"修改失败");
		}
	}
	
	/**
	 * 查询模板的中的品牌下拉框
	 * @return
	 */
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList(){
		List<Map> selectOptionList = brandService.selectOptionList();
		return selectOptionList;
	}
}

File: pinyougou_manager_web/src/main/webapp/js/controller/brandController.js
	//读取列表数据绑定到表单中
	app.controller('brandController', function($scope, $http,brandService,$controller) {
		//继承baseController
		$controller('baseController',{$scope:$scope})
		//查询所有，不分页
 		$scope.findAll = function() {
			brandService.findAll().success(function(response) {
				$scope.list = response;
			});
		} 

		 //品牌增加 or update
		$scope.save = function() {
			//服务层对象
			var serviceObject;
			//因为修改和增加在一个页面中，所以需要判断是否传入了id值，修改需要传id值进行回显
			if ($scope.entity.id != null) {
				serviceObject = brandService.update($scope.entity)
			}else{
				serviceObject = brandService.add($scope.entity)
			}
			serviceObject.success(
				function(response){
					if(response.success){
						$scope.reloadList();
					}else{
						alert(response.message);
					}
				}
			);
		}

		//根据id查询单个品牌信息
		$scope.findById = function(id) {
			brandService.findById(id).success(
					function(response) {
						$scope.entity = response;
					})
		}
		
		 $scope.delete = function(){
		 
		 if(confirm('确定删除？')){
		 	brandService.delete($scope.selectIds).success(
		 		function(response){
		 			if(response.success){
		 			$scope.reloadList();
		 			}else{
		 			alert(response.message);
		 			}
		 		}
		 	);
		 }
		 }
		 
		 //定义搜索对象
		 $scope.searchEntity={};
		 
		 //条件查询
		 $scope.search = function(currentPage,pageSize){
		 	 $http.post('../brand/search.do?currentPage='+currentPage+"&pageSize="+pageSize, $scope.searchEntity).success(
		 	 
		 	 function(response){
		 	 	$scope.paginationConf.totalItems=response.total;//总记录数  
    			$scope.list=response.rows;//给列表变量赋值 
		 	 }
		 	 
		 	 )
		 	 
		 	
		 }
	});